tracing-error/src/backtrace.rs

```diff
@@ -120,6 +120,38 @@ impl SpanTrace {
             }
         });
     }
+
+    /// Returns the status of this SpanTrace, indicating whether this backtrace request was unsupported, empty, or a span trace was actually captured.
+    pub fn status(&self) -> SpanTraceStatus {
+        if self.span == Span::none() {
+            SpanTraceStatus::Empty
+        } else {
+            let mut status = None;
+            self.span.with_subscriber(|(_, s)| {
+                if s.downcast_ref::<WithContext>().is_some() {
+                    status = Some(SpanTraceStatus::Captured);
+                }
+            });
+
+            status.unwrap_or(SpanTraceStatus::Unsupported)
+        }
+    }
+}
+
+/// The current status of a SpanTrace, indicating whether it was captured or whether it is empty
+/// for some other reason.
+#[derive(Debug, PartialEq, Eq)]
+pub enum SpanTraceStatus {
+    /// Formatting a SpanTrace is not supported, likely because there is no ErrorLayer or the
+    /// ErrorLayer is from a different version of tracing_error
+    Unsupported,
+
+    /// The SpanTrace is empty, likely because it was captured outside of any `span`s
+    Empty,
+
+    /// A span trace has been captured and the `SpanTrace` should print reasonable information when
+    /// rendered.
+    Captured,
 }
```

Comment:
let's not expose a public enum, since this is a forwards-compatibility hazard. since enum matching is exhaustive, adding new variants to this is a breaking change. 

there are two solutions to exposing this in a public API without introducing that hazard: 
1. we make the enum private and wrap it in a struct with associated consts, as we do elsewhere, such as `Level`:
  https://github.com/tokio-rs/tracing/blob/77e589906df9702a11f7c7bdb6ea69a3437a320f/tracing-core/src/metadata.rs#L241-L262
2. we put the `#[non_exhaustive]` attribute on the enum (which requires bumping MSRV to 1.40, which isn't compatible w/ our last-three-versions MSRV policy).